<div style="text-align:center; color: blue"><h1>Sprint 5: Hipotesis Testing</h1></div>

In [1]:
import pandas as pd

In [2]:
#importamos los datos
df = pd.read_csv('Lligues_europees_UEFA.csv', encoding='ISO-8859-1', sep=';')
#convertimos a utf-8 para que no haya problemas con los acentos
df = df.apply(lambda x: x.astype(str).str.encode('utf-8').str.decode('utf-8'))
df

Rk            Squad Country LgRk  MP   W   D   L  GF  GA  ... Pts Pts/G   
0    1  Manchester City     ENG    1  37  28   6   3  96  24  ...  90  2.43  \
1    2        Liverpool     ENG    2  36  26   8   2  89  24  ...  86  2.39   
2    3      Real Madrid     ESP    1  37  26   7   4  80  31  ...  85   2.3   
3    4    Bayern Munich     GER    1  34  24   5   5  97  37  ...  77  2.26   
4    5        Paris S-G     FRA    1  37  25   8   4  85  36  ...  83  2.24   
..  ..              ...     ...  ...  ..  ..  ..  ..  ..  ..  ...  ..   ...   
93  94         Bordeaux     FRA   20  37   5  13  19  48  89  ...  28  0.76   
94  95          Venezia     ITA   20  37   6   8  23  34  69  ...  26   0.7   
95  96          Watford     ENG   19  37   6   5  26  33  75  ...  23  0.62   
96  97     Norwich City     ENG   20  37   5   7  25  23  79  ...  22  0.59   
97  98   Greuther Fürth     GER   18  34   3   9  22  28  82  ...  18  0.53   

      xG   xGA    xGD xGD/90     Last 5 Attendance          Top Team Scorer   
0   86.1  26.8   59.3    1.6  W W W W D      52739     Kevin De Bruyne - 15  \
1   84.6  33.1   51.4   1.43  W W W D W      53367       Mohamed Salah - 22   
2   73.0  45.8   27.2   0.73  W W L W D      40624       Karim Benzema - 27   
3   88.1  37.1   51.0    1.5  W W L D D      33176  Robert Lewandowski - 35   
4   71.6  38.1   33.4    0.9  W D D D W      41188       Kylian Mbappé - 25   
..   ...   ...    ...    ...        ...        ...                      ...   
93  41.3  68.5  -27.2  -0.73  D L L L D      22083         Hwang Ui-jo - 11   
94  35.4  70.5  -35.1  -0.95  L L L W D       6705         Thomas Henry - 9   
95  37.8  64.1  -26.3  -0.71  L L L D L      20614     Emmanuel Dennis - 10   
96  33.0  73.3  -40.4  -1.09  L L L L D      26826         Teemu Pukki - 11   
97  30.7  60.3  -29.6  -0.87  D L D L L       7672      Branimir Hrgota - 9   

          Goalkeeper  
0            Ederson  
1            Alisson  
2   Thibaut Courtois  
3       Manuel Neuer  
4       Keylor Navas  
..               ...  
93     Benoît Costil  
94     Sergio Romero  
95        Ben Foster  
96          Tim Krul  
97   Sascha Burchert  

[98 rows x 21 columns]

**EJERCICIO 1**: Descarga el dataset adjunto de datos oficiales de la UEFA y selecciona un atributo del conjunto de datos. Calcula el p-valor y di si rechaza la hipótesis nula cogiendo un alfa de 5%.

- Vamos a formular la hipótesis nula $H_0$. Para ello, seleccionaremos la variable GF (Goles a Favor).
  - $H_0$: El promedio de goles a favor de todos los equipos es 50.
- La hipótesis alternativa $H_1$ será:
  - $H_1$: El promedio de goles a favor de todos los equipos no es 50.

**Seleccionamos la prueba estadística a utilizar**: En este caso, como trabajamos con un promedio y una muestra, podemos usar la prueba t de una muestra.

**Tomamos una decisión**:
- Si el p-valor $< \alpha$ (0.05), rechazamos la hipótesis nula.
- Si el p-valor $> \alpha$ (0.05), no rechazamos la hipótesis nula.


In [3]:
#cargamos la prueba t de student para una muestra
from scipy.stats import ttest_1samp
import numpy as np

#Datos de la columna GF
data = df['GF']

#convertimos los datos a float
data = data.astype(float)
#realizamos la prueba t de student para una muestra, de acuerdo a la hipótesis planteada, la media es 50
tset, pval = ttest_1samp(data, 50)
#imprimimos el p-value
print("p-value: ",np.round(pval,3))

p-value:  0.591


**Finalmente, tomamos una decisión**:
- el p-value > 0.05, por lo que no rechazamos la hipótesis nula.

Esto significa que no hay suficiente evidencia para rechazar la hipótesis nula, por lo que no podemos decir que el promedio de goles a favor de todos los equipos es diferente de 50.


**EJERCICIO 2** : Con el mismo dataset selecciona otros dos atributos del conjunto de datos. Calcula los p-valores y di si rechazan la hipótesis nula cogiendo un alfa de 5%.

Elegiré "GA" (Goles En Contra) y "Pts" (Puntos).

Para "GA" (Goles En Contra):
**Hipótesis**:

- $H_0$: El promedio de goles en contra de todos los equipos es 40.
- $H_1$: El promedio de goles en contra de todos los equipos no es 40.

Para "Pts" (Puntos):
**Hipótesis**:

- $H_0$: El promedio de puntos de todos los equipos es 60.
- $H_1$: El promedio de puntos de todos los equipos no es 60.

Realizaremos las pruebas t de una muestra para ambos atributos y luego compararemos los p-valores obtenidos con el nivel de significancia $\alpha = 0.05$.


In [27]:
#iniciamos con la variable GA
data = df['GA']

#convertimos los datos a float
data = data.astype(float)

#realizamos la prueba t de student para una muestra, de acuerdo a la hipótesis planteada, la media es 40
tset, pval = ttest_1samp(data, 40)
#imprimimos el p-value
print("p-value: ",pval)

p-value:  4.423330192258983e-12


In [28]:
#AHora con la variable Pts
data = df['Pts']

#convertimos los datos a float
data = data.astype(float)

#realizamos la prueba t de student para una muestra, de acuerdo a la hipótesis planteada, la media es 60
tset, pval = ttest_1samp(data, 60)
#imprimimos el p-value
print("p-value: ",pval)

p-value:  9.308715556871636e-09


Dado que nuestro nivel de significancia es:
$ \alpha = 0.05 $

Para "GA" (Goles En Contra):

- $ p $-value: 
$ 4.4233 \times 10^{-12} $ 
$\rightarrow$ Dado que $ 4.4233 \times 10^{-12} $ es mucho menor que 0.05, rechazamos la hipótesis nula $ H_0 $. Esto indica que hay evidencia suficiente para decir que el promedio de goles en contra de todos los equipos no es 40.

Para "Pts" (Puntos):

- $ p $-value: 
$ 9.3087 \times 10^{-9} $
$\rightarrow$ Dado que $ 9.3087 \times 10^{-9} $ es mucho menor que 0.05, también rechazamos la hipótesis nula $ H_0 $. Esto sugiere que hay evidencia suficiente para decir que el promedio de puntos de todos los equipos no es 60.

Ambos p-values son extremadamente pequeños, lo que indica que los resultados son altamente significativos. Es decir, las diferencias observadas en los datos son muy poco probable que se deban al azar.


**EJERCICIO 3**: Continúa con el conjunto de datos adjunto y selecciona tres atributos del conjunto de datos. Calcula el p-valor y di si rechaza la hipótesis nula cogiendo un alfa de 5%.

Elegiremos "xG" (Expectativa de Goles), "xGA" (Expectativa de Goles En Contra) y "xGD" (Diferencia de Expectativa de Goles).

Para "xG" (Expectativa de Goles):\
**Hipótesis**:

- $H_0$: El promedio de expectativa de goles de todos los equipos es 50.
- $H_1$: El promedio de expectativa de goles de todos los equipos no es 50.

Para "xGA" (Expectativa de Goles En Contra):\
**Hipótesis**:

- $H_0$: El promedio de expectativa de goles en contra de todos los equipos es 40.
- $H_1$: El promedio de expectativa de goles en contra de todos los equipos no es 40.

Para "xGD" (Diferencia de Expectativa de Goles):\
**Hipótesis**:

- $H_0$: El promedio de diferencia de expectativa de goles de todos los equipos es 10.
- $H_1$: El promedio de diferencia de expectativa de goles de todos los equipos no es 10.
\
### Una vez obtenidos los p-values, comparamos cada p-value con el nivel de significancia $\alpha = 0.05$.


In [6]:
#primera hipótesis: el promedio de expectativa de goles de todos los equipos es 50

data = df['xG']
data = data.astype(float)
tset, pval = ttest_1samp(data, 50)
print("p-value: ",pval)

p-value:  0.28697674502528275


**Análisis**: Dado que p-value > 0.05, no rechazamos la hipótesis nula. Esto significa que, basados en nuestra muestra de datos, no tenemos suficiente evidencia para deicr que el promedio de expectativa de goles de todos los equipos es diferente de 50.

In [7]:
data = df['xGA']
data = data.astype(float)
tset, pval = ttest_1samp(data, 40)
print("p-value: ",pval)

p-value:  7.647419577409634e-14


**Análisis**: En este caso, dado que puestro p-value es mucho menor que 0.05, rechazamos la hipótesis nula $H_0$​. Esto indica que hay suficiente evidencia para decir que el promedio de expectativa de goles en contra de todos los equipos no es 40.

In [9]:
data = df['xGD']
data = data.astype(float)
tset, pval = ttest_1samp(data, 10)
print("p-value: ",pval)

p-value:  3.075309318326299e-06


**Análisis**: Dado que el p-value es mucho menor que 0.05, rechazamos la hipótesis nula $H_0$. Esto quiere decir que hay sufiente evidencia para decir que el promedio de diferencia de expectativa de goles de todos los equipos no es 10.